controllers/hvpa_controller.go

```diff
@@ -527,7 +527,7 @@ func (r *HvpaReconciler) scaleIfRequired(hpaStatus *autoscaling.HorizontalPodAut
 		log.Error(err, "Error in getting weight based requests in new deployment", "hvpa", hvpa.Namespace+"/"+hvpa.Name)
 	}
 
-	hpaStatus, err = getWeightedReplicas(hpaStatus, hvpa, currentReplicas, 100-vpaWeight, blockedScaling)
+	hpaStatus, err = getWeightedReplicas(hpaStatus, hvpa, currentReplicas, hpaWeight, blockedScaling)
```

Comment:
`hpaWeight` here would have been `100-vpaWeight` but now might be `MaxWeight`. Is this ok?